### ISSUE TITLE
`MainWindow` and `Container` exhibit assymetric behavior on `.close()`

### ISSUE THREAD

**Issue by bruno-f-cruz** [pyapp-kit/magicgui#685]
**Describe the bug**
First of all, I apologize if this is indeed the intended behavior. 
When calling `close` on `MainWindow` and `Container` I am getting different outcomes. `Container` closes the UI element and exits the context, while `MainWindow` closes the inner element but not the OS window.

**To Reproduce**

```python
from magicgui import event_loop
from magicgui.widgets import Button, MainWindow, Container

button = Button(text="Submit")
container = Container(widgets=[button])
# container = MainWindow(widgets=[button])  # Replace me

button.changed.connect(container.close)

with event_loop() as loop:
    container.show()
```

**Expected behavior**
The `close` behavior to be symmetric. Otherwise, is there an intended alternative to close the context from `MainWindow`?

**Environment (please complete the following information):**
 - OS: Win11
 - backend: [e.g. Qt6]
 - magicgui version 0.10.0

**Comment by tlambert03** [comment]
yeah, this is a bug... not the intended behavior.  It occurs because the `_qtpy.widgets.MainWindow` object *has* a container... rather than *is* a container, but that should be fixed.  I agree with your expected behavior.
